aws/resource_aws_neptune_parameter_group.go

```diff
@@ -0,0 +1,266 @@
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+				StateFunc: func(val interface{}) string {
+					return strings.ToLower(val.(string))
+				},
+			},
+			"family": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "Managed by Terraform",
+			},
+			"parameter": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"value": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"apply_method": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "immediate",
+							// this parameter is not actually state, but a
+							// meta-parameter describing how the RDS API call
+							// to modify the parameter group should be made.
+							// Future reads of the resource from AWS don't tell
+							// us what we used for apply_method previously, so
+							// by squashing state to an empty string we avoid
+							// needing to do an update for every future run.
+							StateFunc: func(interface{}) string { return "" },
+						},
+					},
+				},
+				Set: resourceAwsNeptuneParameterHash,
+			},
+		},
+	}
+}
+
+func resourceAwsNeptuneParameterGroupCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).neptuneconn
+
+	createOpts := neptune.CreateDBParameterGroupInput{
+		DBParameterGroupName:   aws.String(d.Get("name").(string)),
+		DBParameterGroupFamily: aws.String(d.Get("family").(string)),
+		Description:            aws.String(d.Get("description").(string)),
+	}
+
+	log.Printf("[DEBUG] Create Neptune Parameter Group: %#v", createOpts)
+	resp, err := conn.CreateDBParameterGroup(&createOpts)
+	if err != nil {
+		return fmt.Errorf("Error creating Neptune Parameter Group: %s", err)
+	}
+
+	d.Partial(true)
+	d.SetPartial("name")
+	d.SetPartial("family")
+	d.SetPartial("description")
+	d.Partial(false)
+
+	d.SetId(*resp.DBParameterGroup.DBParameterGroupName)
+	log.Printf("[INFO] Neptune Parameter Group ID: %s", d.Id())
+
+	return resourceAwsNeptuneParameterGroupUpdate(d, meta)
+}
+
+func resourceAwsNeptuneParameterGroupRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).neptuneconn
+
+	describeOpts := neptune.DescribeDBParameterGroupsInput{
+		DBParameterGroupName: aws.String(d.Id()),
+	}
+
+	describeResp, err := conn.DescribeDBParameterGroups(&describeOpts)
+	if err != nil {
+		return err
+	}
+
+	if len(describeResp.DBParameterGroups) != 1 ||
```

Comment:
While we are trying to prevent errors/panics here we should check `describeResp == nil` as well in the (unlikely) scenario that it comes back empty from the API without an error.